src/patternfly/components/Table/table-grid.scss

```diff
@@ -316,33 +314,36 @@
       top: 0;
       bottom: 0;
       left: 0;
-      width: var(--pf-c-table__expandable-row--before--Width);
       content: "";
+      border: 0;
+      border-left: var(--pf-c-table--tbody--after--BorderColor) solid var(--pf-c-table--tbody--after--BorderLeftWidth);
     }
 
     &.pf-m-expanded {
-      &::after {
-        background-color: var(--pf-c-table__expandable-row--before--BackgroundColor);
-      }
+      --pf-c-table--tbody--after--BorderLeftWidth: var(--pf-c-table--tbody--after--BorderWidth);
+    }
+
+    > tr > * {
+      overflow: hidden;
     }
 
     > tr > :first-child:not(.pf-c-table__check)::before {
+      --pf-c-table__expandable-row--before--BorderLeftWidth: 0;
+
       position: static;
       width: auto;
       background-color: transparent;
     }
   }
-  // Compound expansion responsive end
-
 
   // Expandable row
   .pf-c-table__expandable-row {
     --pf-c-table-cell--responsive--PaddingTop: 0;
     --pf-c-table-cell--responsive--PaddingRight: 0;
     --pf-c-table-cell--responsive--PaddingBottom: 0;
     --pf-c-table-cell--responsive--PaddingLeft: 0;
-    --pf-c-table-cell--PaddingRight: 0;
-    --pf-c-table-cell--PaddingLeft: 0;
+    --pf-c-table--cell--PaddingRight: 0;
+    --pf-c-table--cell--PaddingLeft: 0;
```

Comment:
as per comment above, if we're going with `pf-c-table--cell` we should update all of the var names